src/main/kotlin/lotto/domain/Rank.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.domain
+
+enum class Rank(val matchCount: Int, val prizeMoney: Int) {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    ELSE(0, 0);
+
+    companion object {
+        fun of(matchCount: Int, matchBonus: Boolean): Rank {
+            var rank = values().firstOrNull { it.matchCount == matchCount } ?: ELSE
+            if (rank == SECOND &&
+                !matchBonus
+            ) {
+                rank = THIRD
+            }
```

Comment:
가독성을 위해 if문의 소괄호 안쪽 구문이 가능한 한 여러줄이 되지 않도록 유지해주세요.

```suggestion
            if (rank == SECOND && !matchBonus) {
                rank = THIRD
            }
```

Reply:
변경했습니다!